nat/traversal/pinger_test.go

```diff
@@ -110,6 +102,56 @@ func TestPinger_Provider_Consumer_Ping_Flow(t *testing.T) {
 	}, time.Second, 10*time.Millisecond)
 }
 
+func TestPinger_PingPeer_N_Connections(t *testing.T) {
+	pingConfig := &PingConfig{
+		Interval: 10 * time.Millisecond,
+		Timeout:  1000 * time.Millisecond,
+	}
+
+	provider := newPinger(pingConfig)
+	consumer := newPinger(pingConfig)
+
+	var pPorts, cPorts []int
+	ports, err := port.NewPool().AcquireMultiple(10)
+	assert.NoError(t, err)
+
+	for i := 0; i < 5; i++ {
+		pPorts = append(pPorts, ports[i].Num())
+		cPorts = append(cPorts, ports[5+i].Num())
+	}
+
+	go consumer.PingPeer("192.168.0.101", cPorts, pPorts, 2, 3)
+	conns, err := provider.PingPeer("127.0.0.1", pPorts, cPorts, 2, 3)
+	assert.NoError(t, err)
+
+	assert.Len(t, conns, 3)
+}
+
+func TestPinger_PingPeer_Too_Many_Connections(t *testing.T) {
+	pingConfig := &PingConfig{
+		Interval: 10 * time.Millisecond,
+		Timeout:  1000 * time.Millisecond,
+	}
+
+	provider := newPinger(pingConfig)
+	consumer := newPinger(pingConfig)
+
+	var pPorts, cPorts []int
+	ports, err := port.NewPool().AcquireMultiple(10)
+	assert.NoError(t, err)
+
+	for i := 0; i < 5; i++ {
+		pPorts = append(pPorts, ports[i].Num())
+		cPorts = append(cPorts, ports[5+i].Num())
+	}
+
+	go consumer.PingPeer("192.168.0.101", cPorts, pPorts, 2, 30)
+	conns, err := provider.PingPeer("127.0.0.1", pPorts, cPorts, 2, 30)
+	assert.EqualError(t, err, "not enough connections")
```

Comment:
Test name is Too_Many_Connections but error assertion is `not enough connections`.

Reply:
Renamed test